src/status_im/contexts/keycard/migrate/events.cljs

```diff
@@ -40,31 +35,27 @@
  (fn [{:keys [db]}]
    (let [key-uid                                (get-in db [:profile/profile :key-uid])
          {:keys [initialized? has-master-key?]} (get-in db [:keycard :application-info])
-         {:keys [masked-phrase pin]}            (get-in db [:keycard :migration])
-         on-failure                             (fn []
-                                                  (rf/dispatch [:keycard/disconnect])
-                                                  (rf/dispatch [:navigate-to
-                                                                :screen/keycard.migrate.fail]))]
+         {:keys [masked-phrase pin]}            (get-in db [:keycard :migration])]
+
      (cond
 
        (not initialized?)
        {:fx [[:keycard/init-card
               {:pin        pin
-               :on-success #(get-application-info-and-continue key-uid)
-               :on-failure on-failure}]]}
+               :on-success #(get-application-info-and-continue key-uid)}]]}
 
        (not has-master-key?)
        {:fx [[:effects.keycard/generate-and-load-key
               {:mnemonic   (security/safe-unmask-data masked-phrase)
                :pin        pin
                :on-success #(get-application-info-and-continue key-uid)
-               :on-failure on-failure}]]}
+               :on-failure #(rf/dispatch [:keycard/on-action-with-pin-error %])}]]}
 
        :else
        {:fx [[:effects.keycard/get-keys
               {:pin        pin
                :on-success #(rf/dispatch [:keycard/migration.convert-to-keycard-profile %])
-               :on-failure on-failure}]]}))))
+               :on-failure #(rf/dispatch [:keycard/on-action-with-pin-error %])}]]}))))
```

Comment:
At some point, when we change the code to use `on-error` instead of `on-failure`, could we also add the extra logic to use `rf/call-continuation`? This way we can just pass the event vector and this function becomes easier to unit test.

Reply:
yes sure https://github.com/status-im/status-mobile/issues/21511